cfme/tests/cloud/test_keypairs.py

```diff
@@ -73,41 +78,55 @@ def test_keypair_crud_with_key(provider, appliance):
 @pytest.mark.tier(3)
 def test_keypair_create_cancel(provider, appliance):
     """ This will test cancelling on adding a keypair
-
-    Steps:
-        * Provide Keypair name.
-        * Select Cloud Provider.
-        * Also delete it.
-
     Polarion:
-        assignee: mnadeem
+        assignee: mmojzis
         casecomponent: WebUI
+        caseimportance: low
         initialEstimate: 1/4h
+        testSteps:
+            1. Cancel creating keypair.
     """
     keypair = appliance.collections.cloud_keypairs.create(
         name="",
         provider=provider,
         cancel=True
     )
+    view = keypair.create_view(KeyPairAllView)
+    assert view.flash.assert_message('Add of new Key Pair was cancelled by the user')
 
-    assert not keypair.exists
+
+@pytest.mark.uncollectif(lambda provider: provider.one_of(OpenStackProvider))
+@pytest.mark.tier(3)
+def test_keypair_create_validation(provider, appliance):
+    """ This will test validating the new of name of the key pair
+    Polarion:
+        assignee: mmojzis
+        casecomponent: WebUI
+        caseimportance: low
+        initialEstimate: 1/4h
+        testSteps:
+            1) Try to add key pair with empty name.
+    """
+    keypair_collection = appliance.collections.cloud_keypairs
+    view = navigate_to(keypair_collection, 'Add')
+    view.fill({'provider': provider})
+    assert not view.form.add.active
 
 
+@pytest.mark.uncollectif(lambda provider: provider.one_of(EC2Provider))
```

Comment:
I changed uncollection to override provider to use only Openstack.
This test won't work for EC2 because of bug.
Also this test doesn't need to test multiple providers as it actually is testing something that doesn't belong to provider, but keypair cannot be added when there is no cloud provider.